### PR TITLE
build: update rollup and plugins version

### PR DESCRIPTION
**Pull request by tiborux**
Updates rollup and plugin versions to the lastest version. We tried to change the rollup-plugin-typescript2 for the official one but we got an error. We tried different things and read a lot of posts but the only solution that we found was to continue with the previous plugin.

EX-5336

### CODE REVIEW

File: build/instrumentation.build.js
Comment (javieri-empathy) [reply build/instrumentation.build.js:63]: Okey, this can't be done yet. The core chunk exports certain utilities, and wrapping it with the polyfills plugin in a closure will create invalid JS, as the import/export tokens must appear at the top level.
The only way I can imagine for this to work is to use the top-level await together with a dynamic import to the polyfills. This should ensure that the rest of the waits to be executed, but [this feature hasn't enough support yet](https://caniuse.com/mdn-javascript_operators_await_top_level).

So when we attack the polyfills task, we either inline them in the `core` chunk, or we have to leave another entry chunk like we do now.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: package.json
Comment (tiborux) [reply package.json:52]: We cannot update cssnano to the latest version. If we update to this version, we need to update the vue plugin to the latest version because in the current version they are using `"postcss": "^7.0.11"` and If we update css and autoprefixer to the latest version, we need to update postcss to version 8, but vue plugin is not working with it.

File: tsconfig.json
Comment (javieri-empathy) [inline tsconfig.json:3]: Lets wait until X PR gets merged, but I'm afraid you are going to have to change this to es2019, and set it to es2020 in the rollup config, like @diegopf did in the other PR.

Comment (javieri-empathy) [reply tsconfig.json:3]: This is going to be far more complicated here, probably we will need to configure babel for webpack, as X-Components are exporting modern code

File: vue.config.js
Comment (javieri-empathy) [inline vue.config.js:28]: This is much easier. Vue CLI already adds a babel loader for JS files. It also exposes a transpileDependencies option which you can configure with `transpileDependencies: ['@empathyco/x-components']` so it transpiles x-components.

File: tests/e2e/webpack.config.js
Comment (javieri-empathy) [inline tests/e2e/webpack.config.js:41]: If I'm not mistaken you can simplify this config. This is only needed to transpile the `forEach` utility which is imported by the support file. There is no CSS imported, but just in case I would add the `.js` extension at the end of the regex, as it is the kind of files that babel has to transpile. Also you don't need to make this a pre-loader, a normal loader is fine.

```suggestion
      {
        test: /@empathyco\/x-components.+\.js$/,
        use: ['babel-loader']
      },```

File: babel.config.js
Comment (javieri-empathy) [inline babel.config.js:2]: Isn't it enough with just transpiling the nullish coalescing operator and the optional chaining? Vue preset will try to do more transformations

```suggestion
```

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Review by javieri-empathy — DISMISSED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Review by LuisMartinez15 — APPROVED** [review]

**Pull request merged**
